### PR TITLE
Added renderBadge function to TabItem

### PR DESCRIPTION
**Pull request by ptomasroos**
This PR allows the user of the TabBar to render its own definition of a Badge instead of locking it down to just text.

### CODE REVIEW

**Comment by ptomasroos** [comment]
This fixes #102

File: src/tab/ExNavigationTabBar.js [hunk 1]
```diff
@@ -61,10 +61,17 @@ export default class ExNavigationTabBar extends React.Component {
...
+
       let badge = null;
-      if (item.badgeText) {
+
+      const { renderBadge, badgeText } = item;
+
+      if (renderBadge) {
+        badge = renderBadge();
+      }
+      else if (badgeText) {
```
Comment (chirag04) [inline src/tab/ExNavigationTabBar.js]: keep closing brace and else on the same line.  `} else if { `

Comment (ptomasroos) [inline src/tab/ExNavigationTabBar.js]: Done

**Comment by chirag04** [comment]
👍

**Comment by satya164** [comment]
FYI I'm planning to migrate this to use https://github.com/react-native-community/react-native-tab-view instead since we already have a dependency on it and it has an implementation of bottom tab bar.

**Comment by ptomasroos** [comment]
Sounds good @satya164 , but until then I see no reason why can't bring in a renderBadge func!

**Comment by satya164** [comment]
@ptomasroos You can, but it just duplicates effort.

**Comment by brentvatne** [comment]
Thanks @ptomasroos! Can you rebase this? Also, it might be useful to pass in some parameters to `renderBadge` like we do for `renderTitle` and `renderIcon` (https://github.com/exponentjs/ex-navigation/blob/master/src/tab/ExNavigationTabBar.js#L57-L66) -- people don't need to use them but it is useful sometimes.

**Comment by ptomasroos** [comment]
All right rebased.

**Comment by ptomasroos** [comment]
And updated signature for renderBadge

**Comment by ptomasroos** [comment]
Will open a new PR.

**Pull request closed without merging**